### PR TITLE
1392: Add "related events" section to Topic pages, using banded-background styling

### PR DESCRIPTION
**Pull request by stevejalim**
This changeset adds a new "related events" section to the Topic page, using the four-in-a-row component that we're adopting in a few other places.

It also adds support for automatically deciding whether certain components on the page should have a tinted background or not, in order to generate an alternate-banding look to the page. 

## Screenshots

(Some, but not all, of the permutations possible in the Topic page, showing the automatic tinted-backgrond logic working)

Screenshot_2020-07-05 AV1 Video

-----

Screenshot_2020-07-05 AV1 Video(1)

-----

Screenshot_2020-07-05 AV1 Video(2)

-----

Screenshot_2020-07-05 AV1 Video(3)

-----

Screenshot_2020-07-05 AV1 Video(4)

Related issue: #1392

## Key changes:

- Modelling update to add Topic.related_events StreamField. This is so we can explicitly link to relevant Events, rather than using the Topic.events property, which gets the most recent Events, which isn't always what we want.
- Support manually specifying whether the four-card-row organism has a tinted background or not (and then ensuring it still does in the Events page's Featured panel)
- Support an optional tinted background on the People organism
- Add Topic.get_section_background_panel_hints helper to provide opinionated config to the template for whether a panel should or should not have a tinted background
- Support alternate tinted/not-tinted banding styling on Topics page. Uses the "tint hints" helper method in the Topic page template to pass configuration into the organisms it uses, if they are present

## How to test

- Code is on its way to staging
- Add some related events to a Topic page and preview/publish
- Play around with the varioous permutations of "Recent work", "Related events" and "Experts" sections being present (ie, having any content) and absent
- Confirm featured events styling (tint panel) is still correct and has a tint
- If there is a People section on the homepage, check that renders appropriately (no tint panel, because automatic tinting has not been applied to the homepage)

### CODE REVIEW

File: src/css/templates/events.scss [hunk 1]
```diff
@@ -1,6 +1,24 @@
 body.events {
-  div.events-extra-content {
-    padding-top: 0px;
-    margin-top: -24px;
+  div.header-strip {
+    margin-bottom: 0px; // to allow for the four-card-row div of featured items
```
Comment (schalkneethling) [inline src/css/templates/events.scss:3]: No units with 0 values

Comment (stevejalim) [reply src/css/templates/events.scss:3]: Ah this is already handled in #1605, on which this PR is built. I should have targetted that as the destination branch to make the diff smaller. Amending the target now. Cheers!

**Comment by stevejalim** [comment]
@valgrimm This isn't yet on stage, because CI is stuck. I'll have to get @limed to unblock the CI job when he's around later

**Comment by stevejalim** [comment]
@valgrimm Staged now - please give it a whirl!

**Comment by valgrimm** [comment]
Works for me, although People does get a tint panel when events and related work are both added
image

That said, Steve can fix the tint thing.

Sandra, knowing that this is dummy content, does this look good to you?
https://developer-portal.stage.mdn.mozit.cloud/topics/voice/
@sandrapersing

**Comment by stevejalim** [comment]
@valgrimm The tinting is deliberate -- i thought we agreed on automatical alternating of panel tints

File: developerportal/apps/events/templates/event.html [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {% extends "base.html" %}
```
Comment (stevejalim) [inline developerportal/apps/events/templates/event.html:1]: This files' changes are from another PR which was based off this one, which I should have held off merging, but didn't... Sorry.

File: developerportal/apps/topics/migrations/0055_topic_relevant_events.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Generated by Django 2.2.14 on 2020-07-03 13:05
```
Comment (stevejalim) [inline developerportal/apps/topics/migrations/0055_topic_relevant_events.py:1]: This files' changes are from another PR which was based off this one, which I should have held off merging, but didn't... Sorry.

Ignore for review

File: developerportal/apps/events/templates/events.html [hunk 4]
```diff
@@ -22,7 +22,7 @@
   {% endif %}
```
Comment (stevejalim) [inline developerportal/apps/events/templates/events.html:22]: This files' changes are from another PR which was based off this one, which I should have held off merging, but didn't... Sorry.

Ignore for review

File: developerportal/apps/events/migrations/0030_add_Extra_Content_panel_to_Event.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Generated by Django 2.2.14 on 2020-07-06 12:26
```
Comment (stevejalim) [inline developerportal/apps/events/migrations/0030_add_Extra_Content_panel_to_Event.py:1]: This files' changes are from another PR which was based off this one, which I should have held off merging, but didn't... Sorry.

Ignore for review

File: developerportal/apps/events/migrations/0031_event_sidebar.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Generated by Django 2.2.14 on 2020-07-06 15:49
```
Comment (stevejalim) [inline developerportal/apps/events/migrations/0031_event_sidebar.py:1]: This files' changes are from another PR which was based off this one, which I should have held off merging, but didn't... Sorry.

Ignore for review

File: developerportal/apps/events/models.py [hunk 7]
```diff
@@ -375,6 +375,51 @@ class Event(BasePage):
         help_text="Optional list of speakers for this event",
     )  # DEPRECATED
 
+    extra_content_panel_title = CharField(
```
Comment (stevejalim) [inline developerportal/apps/events/models.py:378]: This files' changes are from another PR which was based off this one, which I should have held off merging, but didn't... Sorry.

Ignore for review

**Comment by valgrimm** [comment]
ah ok, I thought we did too, but then I read your test instructions as people shouldnt be tinted for some reason? Apologies.

This really does look perfect to me, we just need Sandra's word and its done.

**Comment by stevejalim** [comment]
@valgrimm 🤦 Ah yeah, i wrote this

> If there is a People section on the homepage, check that renders appropriately (no tint panel, because automatic tinting has not been applied to the homepage)

thinking only of the homepage, but without specifying that the Topic page would have it tinted if there were enough sections above it.

File: developerportal/apps/topics/tests/test_topics.py [hunk 8]
```diff
@@ -38,6 +38,130 @@ def test_topic_page_articles(self):
...
+                "setup": {
+                    "recent_work": [("heading", "DUMMY")],
+                    "relevant_events": [("heading", "DUMMY")],
+                    "experts": [],
+                },
+                "expected": {
+                    # Whether to use a tinted panel
+                    "recent_work": True,
+                    "relevant_events": False,
+                    "experts": False,  # second item only so no need for tint panel
```
Comment (stevejalim) [inline developerportal/apps/topics/tests/test_topics.py:122]:
```suggestion
                    "experts": False,
```

File: developerportal/apps/topics/tests/test_topics.py [hunk 9]
```diff
@@ -38,6 +38,130 @@ def test_topic_page_articles(self):
...
+                "desc": "Recent work and Relevant events present",
+                "setup": {
+                    "recent_work": [("heading", "DUMMY")],
+                    "relevant_events": [("heading", "DUMMY")],
+                    "experts": [],
+                },
+                "expected": {
+                    # Whether to use a tinted panel
+                    "recent_work": True,
+                    "relevant_events": False,
```
Comment (stevejalim) [inline developerportal/apps/topics/tests/test_topics.py:121]:
```suggestion
                    "relevant_events": False,  # second item only so no need for tint panel
```

**Pull request merged**